pepper-apis/src/main/java/org/broadinstitute/ddp/route/AdminCreateUserLoginAccountRoute.java

```diff
@@ -0,0 +1,178 @@
+            if (listResult.hasThrown() || listResult.hasError()) {
+                Exception e = listResult.hasThrown() ? listResult.getThrown() : listResult.getError();
+                LOG.warn("Error getting client connections for auth0 client {}", auth0ClientId, e);
+                throw ResponseUtil.haltError(HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                        new ApiError(ErrorCodes.SERVER_ERROR, "Error looking up client"));
+            }
+
+            Connection dbConn = listResult.getBody().stream()
+                    .filter(conn -> conn.getStrategy().equals(Auth0ManagementClient.DB_CONNECTION_STRATEGY))
+                    .findFirst()
+                    .orElse(null);
+            if (dbConn == null) {
+                LOG.error("Client {} does not have database connection", auth0ClientId);
+                throw ResponseUtil.haltError(HttpStatus.SC_BAD_REQUEST,
+                        new ApiError(ErrorCodes.OPERATION_NOT_ALLOWED, "Client has no database connection"));
+            } else {
+                LOG.info("Using database connection with name {} for client {}", dbConn.getName(), auth0ClientId);
+            }
+
+            // Create auth0 user
+            String email = payload.getEmail();
+            String randomPassword = GuidUtils.randomPassword();
+            var userResult = auth0Mgmt.createAuth0User(dbConn.getName(), email, randomPassword);
+            if (userResult.hasThrown() || userResult.hasError()) {
+                Exception e = userResult.hasThrown() ? userResult.getThrown() : userResult.getError();
+                LOG.error("Error creating new auth0 account for user {}", user.getGuid(), e);
+                throw ResponseUtil.haltError(HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                        new ApiError(ErrorCodes.SERVER_ERROR, "Error setting up login account"));
+            } else {
+                LOG.info("Created auth0 account for user {}", user.getGuid());
+            }
+
+            // Link auth0 user id
+            var auth0User = userResult.getBody();
+            userDao.assignAuth0UserId(user.getGuid(), auth0User.getId());
+            LOG.info("Assigned auth0 user id to user {}", user.getGuid());
+
+            // Create password reset ticket
+            var ticketResult = auth0Mgmt.createPasswordResetTicket(auth0User.getId(), payload.getRedirectUrl());
+            if (ticketResult.hasThrown() || ticketResult.hasError()) {
+                Exception e = ticketResult.hasThrown() ? ticketResult.getThrown() : ticketResult.getError();
+                LOG.error("Error creating password reset ticket for user {}", user.getGuid(), e);
+                throw ResponseUtil.haltError(HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                        new ApiError(ErrorCodes.SERVER_ERROR, "Error setting up login account"));
+            } else {
+                LOG.info("Created password reset ticket for user {}", user.getGuid());
+            }
+
```

Comment:
Consider moving some of the Auth0 interactions to separate component. This is a lot to do from a request handler. Will be easier to test and/or mock too.

Reply:
Refactored a substantial potion, can refactor more later.